### ISSUE TITLE
config files unnecessarily contain references to TrilinosSSConfig.cmake

### ISSUE THREAD

**Issue by nschloe** [trilinos/Trilinos#2106]
When installing Trilinos, one finds that several package config files, e.g.,
```
/usr/lib/*/cmake/ML/MLConfig.cmake
```
contain the line
```
INCLUDE("${CMAKE_CURRENT_LIST_DIR}/../TrilinosSS/TrilinosSSConfig.cmake")
```
even though the packages themselves do _not_ directly depend on TrilinosSS. This leads to configuration failures on Debian since TriliinosSS is installed only if it is a dependency.

**Comment by mhoemmen** [comment]
@trilinos/framework

**Comment by bmpersc** [comment]
@nschloe that is odd that ML is pulling that in if TrilinosSS is not enabled at all. That should not be happening for sure. I'm less sure if that should not be included if TrilinosSS is enabled, but ML was set to not use it. But if that were the case I suspect you would not have noticed this since the TrilinosSS configure would have been available. By any chance do you still have the configure output from the build that showed this behavior? I'm curious to see what it thought should be enabled.

**Comment by nschloe** [comment]
@bmpersc Any configuration with TrilinosSS that enables any of the affected packages (e.g., ML) will exhibit this behavior.

**Comment by jwillenbring** [comment]
@bmpersc Please take a look at this.

**Comment by bmpersc** [comment]
@nschloe, I'm confused, you say this happens when TrilinosSS is enabled, but ML shows this error when TrilinosSS is not installed because "TriliinosSS is installed only if it is a dependency." If TrilinosSS is enabled then it really should be a dependency of ML. In fact for ML I believe that TrilinosSS is almost always needed. Amesos has a required dependency on TrilinosSS and ML has an optional dependency on Amesos, which is almost always enabled for the use cases for ML that I've seen. So ML does depend on TrilinosSS and this is not a problem that it is trying to pull in the TrilinosSS configure.

The problem with me just trying to replicate this is that if I enable ML and Amesos or just explicitly enable TrilinosSS then I know exactly why TrilinosSS is being enabled and used in ML. What I need to figure out is why at some level you believe that TrilinosSS should not have been enabled and why the configure does believe TrilinosSS needs to be enabled. I've given one example of why that would be above, ML and Amesos are enabled. But, I don't know if that is what you were trying to configure. 

Another thought that I'm not sure what we'd be able to do about this is if the debian install was just not installing TrilinosSS even though it does install Amesos. That would be an issue with debian's understanding of the dependencies. This is just speculation though. I know next to nothing about the debian RPMs that you've setup.

**Comment by nschloe** [comment]
> I'm confused, you say this happens when TrilinosSS is enabled, but ML shows this error when TrilinosSS is not installed because "TriliinosSS is installed only if it is a dependency."

You're right, when building/installing Trilinos as one big thing, there is no problem. The issue arises when splitting Trilinos into its subpackages after the build package (like Debian does). When building with TrilinosSS and ML (which I'll use as an example here), there's no link between the libraries of ML and TrilinosSS, so in Debian it won't be picked up as a dependency. The config file of ML, however, _does_ depend on the config file of TrilinosSS. This results in installation errors in Debian.

In short: The issue is that ML doesn't depend on Trilinos, yet the config file includes the TrilinosSS config.
